src/util/simplify_expr2.cpp

```diff
@@ -350,6 +362,14 @@ struct Addtor
     const std::function<bool(const expr2tc &)> &is_constant,
     std::function<constant_type &(expr2tc &)> get_value)
   {
+    // Is a vector operation ? Apply the op
+    if(is_constant_vector2t(op1) || is_constant_vector2t(op2))
+    {
+      std::function<expr2tc(type2tc, expr2tc, expr2tc)> op =
+        [](type2tc t, expr2tc e1, expr2tc e2) { return add2tc(t, e1, e2); };
+      return distribute_vector_operation(op, op1, op2);
```

Comment:
This and the below calls would benefit from distribute_vector_operation() being a function template.